File: Go/Tournament Winner/solution_1.go
package main

const HOME_TEAM_WON = 1

func TournamentWinner(competitions [][]string, results []int) string {

	currentBestTeam := ""
	scores := map[string]int{currentBestTeam: 0}

	for i, competition := range competitions {

		result := results[i]
		homeTeam, awayTeam := competition[0], competition[1]

		winner := awayTeam

		if result == HOME_TEAM_WON {
			winner = homeTeam
		}

		updateScores(winner, 3, scores)

		if scores[winner] > scores[currentBestTeam] {
			currentBestTeam = winner
		}

	}

	return currentBestTeam

}

func updateScores(team string, points int, scores map[string]int) {

	scores[team] += points
}

File: Python/Easy/Tournament Winner/solution_1.py
HOME_TEAM_WON = 1

def tournamentWinner(competitions, results):
    currentWinner = ""
    dic = {currentWinner : 0} 

    for i , competition in enumerate(competitions):
        result = results[i]
        homeTeam , awayTeam = competition

        winner = homeTeam if result == HOME_TEAM_WON else awayTeam

        updateScores(winner , 3 , dic)

        if dic[winner] > dic[currentWinner]:
            currentWinner = winner

    return currentWinner

def updateScores(team, points, hashTable):

    if team not in hashTable:
        hashTable[team] = 0
    
    hashTable[team] += points

